src/game/Unit.cpp

```diff
@@ -218,9 +218,16 @@ Unit::Unit() :
 
     for (int i = 0; i < MAX_ATTACK; ++i)
     {
-        m_weaponDamage[i][MINDAMAGE] = BASE_MINDAMAGE;
-        m_weaponDamage[i][MAXDAMAGE] = BASE_MAXDAMAGE;
+        for (int j = 0; j < MAX_ITEM_PROTO_DAMAGES; j++)
+        {
+            m_weaponDamage[i][j].damage[MINDAMAGE] = (j == 0) ? BASE_MINDAMAGE : 0;
+            m_weaponDamage[i][j].damage[MAXDAMAGE] = (j == 0) ? BASE_MAXDAMAGE : 0;
+            m_weaponDamage[i][j].school = SPELL_SCHOOL_NORMAL;
```

Comment:
It is simple to fill a basic array. If a player takes up weapon with elemental damage, the damage it will be added into the correct slot with the necessary parameters